app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -64,120 +50,63 @@ export function createPPOMMiddleware<
   ppomController: PPOMController,
   preferencesController: PreferencesController,
   networkController: NetworkController,
-  // TODO: Replace `any` with type
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  appStateController: any,
-  updateSecurityAlertResponseByTxId: (
-    req: JsonRpcRequest<JsonRpcParams> & {
-      securityAlertResponse: SecurityAlertResponse;
-    },
+  appStateController: AppStateController,
+  updateSecurityAlertResponse: (
+    method: string,
+    signatureAlertId: string,
     securityAlertResponse: SecurityAlertResponse,
   ) => void,
 ) {
   return async (
-    req: JsonRpcRequest<Params> & {
-      securityAlertResponse: SecurityAlertResponse;
-    },
+    req: JsonRpcRequest<Params>,
     _res: JsonRpcResponse<Result>,
     next: () => void,
   ) => {
     try {
       const securityAlertsEnabled =
         preferencesController.store.getState()?.securityAlertsEnabled;
+
       const { chainId } = networkController.state.providerConfig;
+
       if (
-        securityAlertsEnabled &&
-        CONFIRMATION_METHODS.includes(req.method) &&
-        SUPPORTED_CHAIN_IDS.includes(chainId)
+        !securityAlertsEnabled ||
+        !CONFIRMATION_METHODS.includes(req.method) ||
+        !PPOM_SUPPORTED_CHAIN_IDS.includes(chainId)
       ) {
-        // eslint-disable-next-line require-atomic-updates
-        const securityAlertId = uuid();
-        let securityAlertResponse: SecurityAlertResponse = {
-          reason: BlockaidResultType.Loading,
```

Comment:
Note that the `reason` and `result_type` properties are using values from the opposite enums, hence why I've had to update the `BlockaidBannerAlert.